File: Priya_Mobile1_Demo/Lec7GridViewDemo_UpdatedFromClass/app/src/main/java/com/example/lec7gridviewdem/SplashActivity.java
package com.example.lec7gridviewdem;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        //we want to set up a timer task, and a timer
        //timer task says what task should be done
        //timer task implements runnable (to create a thread for performing a task

        TimerTask timerTask = new TimerTask() {
            @Override
            public void run() {
                //start the next task
                startActivity(new Intent(SplashActivity.this,
                                        MainActivity.class));
                //finish the current activity object
                finish();
            }
        };

        //time specifies the delay and the task to execute after the delay (ms)
        Timer timer = new Timer();

        //schedule the timer
        timer.schedule(timerTask,3000); //delay is in milliseconds
    }
}

File: 2021 Special Topic Web Mobile App/PocketBook_Version2/app/src/main/java/com/example/pocketbook/SplashActivity.kt
package com.example.pocketbook

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import java.util.*

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val task: TimerTask = object : TimerTask() {
            override fun run() {
                finish()
                startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
            }
        }
        val opening = Timer()
        opening.schedule(task, 1500)
    }
}